tests/lolly/data/lolly_string_test.cpp

```diff
@@ -69,3 +69,27 @@ TEST_CASE ("test append") {
   str << string_u16 (u"yz");
   CHECK_EQ (str == string_u16 (u"xyz"), true);
 }
+
+TEST_CASE ("test reserve along with append") {
+
+  SUBCASE ("reserved more space") {
+    auto str= string_u16 ();
+    str->reserve (6);
+    str << u'x';
+    CHECK_EQ (str == u"x", true);
+    str << string_u16 (u"yz");
+    CHECK_EQ (str == u"xyz", true);
+    str << string_u16 (u": larger than reserved space");
+    CHECK_EQ (str == u"xyz: larger than reserved space", true);
+  }
+  SUBCASE ("reserved the same space") {
+    auto str= string_u16 (u"abc");
+    str->reserve (3);
+    CHECK_EQ (str == u"abc", true);
+  }
+  SUBCASE ("reserved less space should take no effect") {
+    auto str= string_u16 (u"abc");
+    str->reserve (2);
+    CHECK_EQ (str == u"abc", true);
+  }
+}
```

Comment:
we need a test case for different n of expand or shrink, expecially for shrinking

Reply:
test is added